roles/nf-core/tasks/main.yml

```
@@ -0,0 +1,72 @@
+---
+
+- name: install nf-core
+  local_action:
+    module: pip
+    virtualenv_command: /lupus/ngi/irma3/virtualenv-15.0.0/virtualenv.py
+    virtualenv: "{{ nf_core_env }}"
+    name: nf-core
+    state: present
+
+- name: Download NextFlow
+  local_action:
+    get_url url="{{ nextflow_download_url }}" dest="{{ nf_core_env }}/bin" mode="u+rwx,g=rwx"
+
+- name: Install NextFlow
+  local_action:
+    module: command
+    _raw_params: "{{ nf_core_env }}/bin/nextflow"
+  environment:
+    NXF_LAUNCHER: "/scratch"
+    NXF_HOME: "{{ nf_core_env }}/nextflow"
+    NXF_WORK: "/scratch"
+
+- name: nf-core
+  command: "{{ nf_core_env }}/bin/nf-core -v download {{ item.name }} --singularity --compress none --outdir {{ sw_path }}{{ item.name }} --release {{ item.release }}"
+  environment:
+    PATH: "{{ ansible_env.PATH }}:{{ nf_core_env }}/bin"
+  args:
+    chdir: "{{ sw_path }}"
+  with_items: "{{ pipelines }}"
+
+- name: Create directories for singularity images
+  file:
+    path: "{{ ngi_containers }}/{{ item.name }}"
+    state: directory
+    mode: g+s
+  with_items: "{{ pipelines }}"
+
+- name: Move singularity images
+  shell: "mv {{ sw_path }}{{ item.0.name }}/singularity-images/{{ item.1 }} {{ ngi_containers }}/{{ item.0.name }}/"
+  with_subelements:
+    - "{{ pipelines }}"
+    - containers
+
+- name: Create nf-core config
+  template:
+    src: "site.config.j2"
+    dest: "{{ ngi_pipeline_conf }}/{{ item.0.name }}_{{ item.1.site }}.config"
+  with_nested:
+  - "{{ pipelines }}"
+  - [ { site: "sthlm", project_id: "{{ ngi_pipeline_sthlm_delivery }}" }, { site: "upps", project_id: "{{ ngi_pipeline_upps_delivery }}" }]
+
+- name: Set alias for pipeline
+  lineinfile:
+    dest: "{{ ngi_pipeline_conf }}/{{ item.1.script }}"
+    line: >
+          alias {{ item.0.name }}='nextflow run {{ sw_path }}{{ item.0.name }}/workflow/main.nf \
```

Don't need the `main.nf` bit.. (in fact, better without)
```suggestion
          alias {{ item.0.name }}='nextflow run {{ sw_path }}{{ item.0.name }}/workflow/ \
```
And just checking: `{{ sw_path }}{{ item.0.name }}` definitely shouldn't have a `/` in the middle?

---

`sw_path` is defined in `host_vars/127.0.0.1/main.yml` with a trailing slash: `sw_path: "{{ root_path }}/sw/"`. So this should work.
Often enough we get double slashes from that - which of course is no problem technically, it just triggers me :grimacing: :smile:

---

Removed the `main.nf` bit. For the `sw_path`, as Stephan said, there was a trailing slash in the original file which I haven't changed.